insights/specs/datasources/yum_updates.py

```diff
@@ -122,21 +125,28 @@ def __init__(self, build_pkgcache=False):
         self.base.doGenericSetup(cache=0 if build_pkgcache else 1)
         self.releasever = self.base.conf.yumvar['releasever']
         self.basearch = self.base.conf.yumvar['basearch']
-        self.packages = []
-        self.repos = []
-        self.updict = {}
+        self.naevr_to_repo = {}
+        self.pkg_tup_to_advisories = {}
 
     @staticmethod
-    def pkg_cmp(a, b):
-        vercmp = a.verCMP(b)
+    def pkg_cmp(a_tup, b_tup):
+        a_n, a_a, a_e, a_v, a_r = a_tup[0]
+        b_n, b_a, b_e, b_v, b_r = b_tup[0]
+        a_repoid = a_tup[1]
+        b_repoid = b_tup[1]
+
+        if a_n != b_n:
+            return -1 if a_n < b_n else 1
+        vercmp = rpmUtils.miscutils.compareEVR((a_e, a_v, a_r,), (b_e, b_v, b_r,))
```

Comment:
is there a func to compare EVRA? It might happen that there will be the same update with `i686` (32bit) and `x86_64` (64bit) arch.  `a_a` and `b_a` seem to be unused for comparison. Comparing arch as string is probably fine, I can take a look how it is handled in patch/vmaas and in rpm LabelCompare